### PR TITLE
Products controller, model, and tests

### PR DESCRIPTION
**Pull request by emaust**

### CODE REVIEW

File: db/product_seeds.csv [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
+Empty Box,This is the perfect starter home! Fits two comfortably. Going fast!,active,90.39,12,https://imgur.com/94eWfts
+Cat Food,Crunchy and dry. Perfect for snacking.,active,3.85,22,https://imgur.com/o9c2ZNz
+Full Bag of Cat Food,This is an unusual and impressive find! Completely unopened and very heavy!,active,250.10,1,https://imgur.com/bdizBTv
+Snack Pack,Includes kibble, one cheese nub, and nice stale fig newtons,active,10.15,19,https://imgur.com/3cEwmVL
+Gnawed Carrot,Very very orange with beautiful teeth marks,active,4.76,50,https://imgur.com/QoFywrz
+Almost Whole Cookie,Excellent texture. Hard as rock!,active,39.31,21,https://imgur.com/H8DS8Kt
+Corn Cob,A work of art! Comes pre-eaten and ready for any activity,active,20.08,22,https://imgur.com/A45AqiT
+Styrofoam Cup,Has straw and lid. Drink not include!!,active,15.2,23,https://imgur.com/Bef6g06
+Half of Dirty Apple,Crisp with pleasant gritty exterior,active,23.96,24,https://imgur.com/6js2kcS
+Whole Egg,Really great smooth and unbroken egg. Ready for a smash!!,active,159.68,1,https://imgur.com/Si1R1jK
```
Comment (mdove92) [inline db/product_seeds.csv:25]: This is exactly what I've been looking for! I'd like to buy all of your stock of whole eggs. Thank you.

File: test/controllers/products_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            price: 12,
+            stock: 14,
+            photo_url: "http://imgur.com/eggshells4life"
+          }
+        }
+        expect { post products_path, params: product_hash }.must_differ 'Product.count', 1
+        must_redirect_to product_path(Product.find_by(name: "trashgarbage").id)
+      end
+
+      it "does not create a new product when given invalid information and redirects to new product path" do
```
Comment (tofuandeve) [inline test/controllers/products_controller_test.rb]:
```suggestion
      it "does not create a new product when given invalid information and renders to new product path" do
```

File: test/controllers/products_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        get edit_product_path(valid_product1.id)
+
+        must_respond_with :success
+      end
+    end
+
+    describe "update" do
+
+      before do
+        @product = {
```
Comment (tofuandeve) [inline test/controllers/products_controller_test.rb:140]: Nit pick: This variable name can be @product_info or @product_hash cause it's not really a product yet.

File: test/controllers/products_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+              name: "regular pumpkin with nibbles on top",
+              status: "active",
+              description: "work of art", 
+              price: 50,
+              stock: 1,
+              photo_url: "http://imgur.com/pumpkinbits"
+            }
+          }
+
+        @invalid_product = {
```
Comment (tofuandeve) [inline test/controllers/products_controller_test.rb:152]: this one too maybe

**Review by mdove92 — APPROVED** [review]
You SMASHED it! I love this. You two did excellent work, and I love all of our data!

**Review by tofuandeve — APPROVED** [review]
I left one or two minor comments if you want to address, but overall this is an awesome PR. I'll just go ahead and approve this. Great work!

**Pull request merged**